sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/quickpulse/QuickPulseConfiguration.java

```diff
@@ -0,0 +1,129 @@
+    private ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics = new ConcurrentHashMap<>();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    public synchronized String getEtag() {
+        return this.etag.get();
+    }
+
+    public synchronized void setEtag(String etag) {
+        this.etag.set(etag);
+    }
+
+    public synchronized ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> getMetrics() {
+        return this.metrics;
+    }
+
+    public synchronized void setMetrics(ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> metrics) {
+        this.metrics = metrics;
+    }
+
+    public synchronized void updateConfig(String etagValue, ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> otelMetrics) {
+        if (!Objects.equals(this.getEtag(), etagValue)){
+            this.setEtag(etagValue);
+            this.setMetrics(otelMetrics);
+        }
+
+    }
+
+    public ConcurrentHashMap<String,  ArrayList<OpenTelMetricInfo>> parseMetrics(HttpResponse response) {
+
+        ConcurrentHashMap<String, ArrayList<OpenTelMetricInfo>> requestedMetrics = new ConcurrentHashMap<>();
+        try {
+
+            String responseBody = response.getBodyAsString().block();
+            if (responseBody == null || responseBody.isEmpty()) {
+                return new ConcurrentHashMap<String,  ArrayList<OpenTelMetricInfo>>();
+            }
+            JsonNode rootNode = objectMapper.readTree(responseBody);
+            JsonNode metricsNode = rootNode.get("Metrics");
+
+            if (metricsNode instanceof ArrayNode) {
+                ArrayNode metricsArray = (ArrayNode) metricsNode;
+                for (JsonNode metricNode : metricsArray) {
+                    OpenTelMetricInfo metric = new OpenTelMetricInfo();
+                    metric.setId(metricNode.get("Id").asText());
+                    metric.setAggregation(metricNode.get("Aggregation").asText());
+                    metric.setTelemetryType(metricNode.get("TelemetryType").asText());
+                    String projection = metricNode.get("Projection").asText();
+                    metric.setProjection(projection);
+                    if (Objects.equals(metricNode.get("TelemetryType").asText(), "Event")) {
+                        int dotIndex = projection.indexOf(".");
+                        if (dotIndex != -1) {
+                            projection = projection.substring(dotIndex + 1);
+                        }
+                    }
+                    metric.setProjection(projection);
+                    requestedMetrics.computeIfAbsent(metric.getProjection(), k -> new ArrayList<>()).add(metric);
+                }
+            }
+            return requestedMetrics;
+        } catch (Exception e) {
+            logger.verbose("Failed to parse metrics from response: %s", e.getMessage());
+        }
+        return new ConcurrentHashMap<String,  ArrayList<OpenTelMetricInfo>>();
```

Comment:
nit: For any object name / var that starts with "OpenTel" > can we rename those to start with "OTel" instead?

Reply:
fixed